kratos/processes/find_global_nodal_entity_neighbours_process.cpp

```diff
@@ -0,0 +1,219 @@
+        auto colors = MPIColoringUtilities::ComputeCommunicationScheduling(send_list, r_data_communicator);
+
+        // finalize computation of neighbour ids on owner nodes
+        NonLocalMapType recv_map;
+        for(int color : colors) {
+            if (color >= 0) {
+                // recev the global neighbours as computed on color
+                recv_map[color] = r_data_communicator.SendRecv(non_local_map[color], color, color);
+
+                for(auto& r_item : recv_map[color]) {
+                    auto& recv_node = mrModelPart.GetNode(r_item.first);
+                    auto& neighbours = recv_node.GetValue(mrOutputVariable);
+                    for(auto& r_gp : r_item.second.GetContainer()) {
+                        neighbours.push_back(r_gp);
+                    }
+                }
+            }
+        } // after this loop is finished neighbours are ok for the owner nodes
+
+        // fill back the recv_map with the updated information
+        for(int color : colors) {
+            if (color >= 0) {
+                for(auto& r_item : recv_map[color]) {
+                    //r_item.first contains the id of the node
+                    //r_item.second contains the list of neighbours
+                    auto& r_recv_node = mrModelPart.GetNode(r_item.first);
+                    r_item.second = r_recv_node.GetValue(mrOutputVariable);
+                }
+
+                //obtain the final list of neighbours for nodes owned on color
+                auto final_gp_map = r_data_communicator.SendRecv(recv_map[color], color, color );
+
+                //update the local database
+                for(auto& r_item : final_gp_map) {
+                    auto& r_recv_node = mrModelPart.GetNode(r_item.first);
+                    r_recv_node.GetValue(mrOutputVariable) = r_item.second;
+                }
+            }
+        }
+    }
+
+    auto constructor_functor =  ComputeNeighbourListFunctor<ModelPart::NodesContainerType, Variable<GlobalEntityPointersVectorType>>(r_nodes, mrOutputVariable);
+
+    GlobalPointerCommunicator<EntityType> pointer_comm(r_data_communicator, constructor_functor);
+    auto id_proxy = pointer_comm.Apply(
+            [](GlobalPointer<EntityType> const& gp){return gp->Id();}
+    );
+
+    block_for_each(r_nodes, [&](NodeType& rNode){
+        auto& neighbours = rNode.GetValue(mrOutputVariable);
+        neighbours.shrink_to_fit();
+        std::sort(neighbours.ptr_begin(), neighbours.ptr_end(),
+            [&id_proxy](GlobalPointer<EntityType> const& gp1, GlobalPointer<EntityType> const& gp2)
+            {
+                return id_proxy.Get(gp1) < id_proxy.Get(gp2);
+            }
+        );
+    });
+
+    KRATOS_CATCH("");
+}
+
+template<class TContainerType>
+void FindGlobalNodalEntityNeighboursProcess<TContainerType>::Clear()
+{
+    auto& rNodes = mrModelPart.Nodes();
+    VariableUtils().SetVariable(mrOutputVariable, GlobalEntityPointersVectorType(), rNodes);

```

Comment:
```suggestion
    VariableUtils().SetNonHistoricalVariable(mrOutputVariable, GlobalEntityPointersVectorType(), rNodes)

```
According to line 80, shouldn't this be in the non-historical database?

Reply:
good catch... may be no one was using `Clear()` until now...

Reply:
Most probably yes...